apollo-federation/src/connectors/json_selection/methods/public/to_string.rs

```diff
@@ -0,0 +1,388 @@
+            span: 0..8,
+        }
+    }
+
+    fn get_shape(args: Vec<WithRange<LitExpr>>, input: Shape) -> Shape {
+        let location = get_location();
+        to_string_shape(
+            &WithRange::new("toString".to_string(), Some(location.span)),
+            Some(&MethodArgs { args, range: None }),
+            input,
+            Shape::none(),
+            &IndexMap::default(),
+            &location.source_id,
+        )
+    }
+
+    #[test]
+    fn to_string_shape_should_return_string_for_any_input() {
+        assert_eq!(
+            get_shape(vec![], Shape::int([])),
+            Shape::string([get_location()])
+        );
+    }
+
+    #[test]
+    fn to_string_shape_should_return_string_for_string_input() {
+        assert_eq!(
+            get_shape(vec![], Shape::string([])),
+            Shape::string([get_location()])
+        );
+    }
+
+    #[test]
+    fn to_string_shape_should_return_string_for_bool_input() {
+        assert_eq!(
+            get_shape(vec![], Shape::bool([])),
+            Shape::string([get_location()])
+        );
+    }
+
+    #[test]
+    fn to_string_shape_should_return_string_for_null_input() {
+        assert_eq!(
+            get_shape(vec![], Shape::null([])),
+            Shape::string([get_location()])
+        );
+    }
+
+    #[test]
+    fn to_string_shape_should_error_on_args() {
+        assert_eq!(
+            get_shape(
+                vec![WithRange::new(LitExpr::String("arg".to_string()), None)],
+                Shape::int([])
+            ),
+            Shape::error(
+                "Method ->toString does not accept any arguments, but 1 were provided".to_string(),
+                [get_location()]
+            )
+        );
+    }
+
+    #[test]
+    fn to_string_shape_should_return_string_for_none_args() {
+        let location = get_location();
+        assert_eq!(
+            to_string_shape(
+                &WithRange::new("toString".to_string(), Some(location.span)),
+                None,
+                Shape::int([]),
+                Shape::none(),
+                &IndexMap::default(),
+                &location.source_id
+            ),
+            Shape::string([get_location()])
+        );
+    }
+
+    #[test]
+    fn to_string_shape_should_error_for_object_input() {
+        assert_eq!(
+            get_shape(vec![], Shape::empty_object([])),
+            Shape::error_with_partial(
+                "Method ->toString cannot convert arrays or objects to strings. Use ->jsonStringify instead"
+                    .to_string(),
+                Shape::none(),
+                [get_location()]
+            )
+        );
+    }
+
+    #[test]
+    fn to_string_shape_should_error_for_array_input() {
+        assert_eq!(
+            get_shape(vec![], Shape::tuple([], [])),
+            Shape::error_with_partial(
+                "Method ->toString cannot convert arrays or objects to strings. Use ->jsonStringify instead"
```

Comment:
```suggestion
                "Method ->toString cannot convert arrays or objects to strings. Use ->jsonStringify or ->joinNotNull instead"
```

I think in some cases, joining to like `a,b,c` is what the user might be looking for.

Complete side note, we should probably start including docs links somehow for some of these errors 🤔. Something to investigate at some point.